lang/compiler.go

```diff
@@ -118,6 +201,133 @@ func (c *ASTCompiler) PrependFile(file *ast.File) {
 	c.AST.Files = append([]*ast.File{file}, c.AST.Files...)
 }
 
+// TableObjectCompiler compiles a TableObject into an executable flux.Program.
+// It is not added to CompilerMappings and it is not serializable, because
+// it is impossible to use it outside of the context of an ongoing execution.
+type TableObjectCompiler struct {
+	Tables *flux.TableObject
+	Now    time.Time
+}
+
+func (c *TableObjectCompiler) Compile(ctx context.Context) (flux.Program, error) {
+	ider := &ider{
+		id:     0,
+		lookup: make(map[*flux.TableObject]flux.OperationID),
+	}
+	spec := &flux.Spec{
+		Now: c.Now,
+	}
+	visited := make(map[*flux.TableObject]bool)
+	buildSpec(c.Tables, ider, spec, visited)
+
+	specCompiler := &SpecCompiler{
+		Spec: spec,
+	}
+	return specCompiler.Compile(ctx)
+}
+
+func (*TableObjectCompiler) CompilerType() flux.CompilerType {
+	panic("TableObjectCompiler is not associated with a CompilerType")
```

Comment:
This method is needed when a `Compiler` is sent over the network, so the receiving end knows how to deserialize it from JSON.  For a REPL in `influxdb` of `idpe`, seems like we'll want to be able to serialize `TableObjectCompiler`?  It's unclear to me how this will work.

Reply:
I think we discussed about this yesterday. Is it ok if I leave like this for now, and when we decide how to serialize TableObjects we change this bit?

Reply:
I think it's fine for now.

It's unclear to me how having a `TableObjectCompiler` works with the changes that @jsternberg has in mind for the interpreter.  Jonathan, do you want to have a look at these changes?